bionic/task_state.py

```diff
@@ -142,13 +145,33 @@ def attempt_to_complete_from_cache(self):
         if not all(axr.can_load() for axr in self._cache_accessors):
             return
 
-        # We only load the hashed result while completing task state
-        # and lazily load the entire result when needed later.
-        value_hashes_by_dnode = {}
-        for accessor in self._cache_accessors:
-            value_hash = accessor.load_result_value_hash()
-            value_hashes_by_dnode[accessor.query.dnode] = value_hash
-        self._result_value_hashes_by_dnode = value_hashes_by_dnode
+        self._load_value_hashes()
+
+    def refresh_all_cache_state(self, bootstrap):
```

Comment:
This makes it sound like you will refresh in-memory cache too. Maybe call this `refresh_persistent_cache`?

Reply:
Yes, good point. We're a bit sloppy with the use of "cache" in a few places.